### ISSUE TITLE
Liquid Comments within <style> and <script> tags

### ISSUE THREAD

**Issue by freshwater-dev** [panoply/vscode-liquid#122]
I have <style> and <script> tags within a .liquid file, but when I go to comment out lines within those tags, it's using Liquid comments (ie, {% comment %}) instead of CSS or JavaScript comments.

See attached screenshot as an example.

Is there a way for comments to be in the native language and not always liquid comments?
<img width="782" alt="liquid comment error" src="https://user-images.githubusercontent.com/113377762/198097192-ada86416-063f-4024-a45e-bba3b822fad9.png">

**Comment by muchisx** [comment]
This seems like a feature that depends on taste.

Liquid comments are useful in a way that independant of the language, they will not polute your production code delivered to the client. So for example in shopify costumers will not encounter your comments when inspecting your code, they will be removed server-side.

In my preference, I much rather that feature, so when I work in a .liquid file I expect liquid comments across the whole file but I understand where you come from.

So maybe an option to override this behavior inside HTML <script> and <style> tags and css.liquid+js.liquid could benefit your type of user.

But given the fact that most effort is being put towards the Liquify extension I doubt this should is worth of a minor change in the current temporary version of the extension

What do you think @panoply ?

**Comment by panoply** [comment]
@muchisx good point. I'll have to investigate further here, it might be simple enough to apply but maybe it is better for Liquify as embedded code regions are handled much more gracefully.

**Comment by panoply** [comment]
On further investigation it seems the Liquid comments are not respecting the grammar injection within embedded code regions. This means highlighting is still being applied to code contained within `{% comment %}`

@freshwater-dev If the Liquid comments respected the syntax coloring, would this help?

**Comment by freshwater-dev** [comment]
Thanks for the quick replies and thanks for this extension, @panoply! It's a massive massive help to Shopify development.

Respecting syntax coloring would be nice, but not a huge impact for me personally.

As for my original comment, my thought is that the type of comment should be native to code you're commenting. So even within a .liquid file, it should be a css comment within a <style> tag and javascript comment within a <script> tag. Without native commenting, it makes it difficult to quickly test different features/options when developing. I understand not everyone might want this as @muchisx raised, but at the very least I feel like it should be an option.

Also, there's something weird going on where the auto-save format forces the liquid comments into new lines, but then when you go to uncomment them, it just adds more comments. This bug doesn't happen when the comments are on one line, so is there a way to prevent the auto-save format from forcing new line breaks?

This is what I want:
`{%- comment -%} {{ product.description }} {%- endcomment -%}`

This is what happens:
`{%- comment -%}
      {{ product.description }}
   {%- endcomment -%}`

Then this is what happens when you go to remove the comment:
`{%- comment -%} {%- comment -%}
          {{ product.description }}
  {%- endcomment -%} {%- endcomment -%}`

**Comment by panoply** [comment]
> Also, there's something weird going on where the auto-save format forces the liquid comments into new lines, 

Ouff... That is a defect in Prettify. I'll need to address it. Try single-line comments, eg: `{% # comment %}`

>  my thought is that the type of comment should be native to code you're commenting

I agree, the logic should be respected based on the code input wrapping, but it might be a little difficult to employ here. I'd need to investigate further.

**Comment by freshwater-dev** [comment]
How can I create single-line comments? Whenever I hit save, the formatter automatically creates line breaks. I don't see any settings to change that.

Thanks!

**Comment by panoply** [comment]
@freshwater-dev working on this one now. The default behaviour for comment block regions is to apply breaks. My thoughts here are to simply exclude breaks whenever there is no whitespace character separating the commented token and opening comment code block, for example:

This will not break onto new lines:

```
.class {
  {% comment %}position: absolute;{% endcomment %}
}
```

This will break onto new lines:

```
.class {
  {% comment %} position: absolute;{% endcomment %}
}
```

In any sense, the next release you'll be able to use Liquid line comments, eg: 

This will break onto new lines:

```
.class {
  {% # position: absolute; %}
}
```

Any suggestions or thoughts are welcome.
